irohad/util/utility_client.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "util/utility_client.hpp"
+
+#include <map>
+
+#include "util/proto_status_tools.hpp"
+
+#include <grpc++/grpc++.h>
+#include <boost/optional.hpp>
+#include "common/bind.hpp"
+#include "logger/logger.hpp"
+#include "network/impl/grpc_channel_builder.hpp"
+#include "utility_endpoint.grpc.pb.h"
+
+using namespace iroha::utility_service;
+
+using iroha::operator|;
+
+struct UtilityClient::StubHolder {
+  StubHolder(const std::string &address)
+      : channel_(grpc::CreateCustomChannel(
+            address,
+            ::grpc::InsecureChannelCredentials(),
+            iroha::network::details::getChannelArguments<
+                proto::UtilityService_v1>())),
+        stub_(proto::UtilityService_v1::NewStub(channel_)) {}
+
+  std::shared_ptr<grpc::Channel> channel_;
+  std::unique_ptr<proto::UtilityService_v1::StubInterface> stub_;
+};
+
+UtilityClient::UtilityClient(std::string const &irohad_address,
+                             logger::LoggerPtr log)
+    : log_(std::move(log)),
+      stub_holder_(std::make_unique<StubHolder>(irohad_address)) {}
+
+UtilityClient::~UtilityClient() = default;
+
+bool UtilityClient::waitForServerReady(
+    std::chrono::milliseconds timeout) const {
+  auto state = stub_holder_->channel_->GetState(true);
+  auto state_is_ready = [&state] {
+    return state == grpc_connectivity_state::GRPC_CHANNEL_READY;
+  };
+  auto deadline = std::chrono::system_clock::now() + timeout;
+  while (not state_is_ready()
+         and deadline > ::std::chrono::system_clock::now()) {
```

Comment:
'steady_clock' will be better here because of monotonic grow. 'system_clock' can be decreased.
https://ru.cppreference.com/w/cpp/chrono/steady_clock

Reply:
maybe we should do similar change in other similar places